python/pyarrow/_s3.pyx

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# cython: language_level = 3
+
+import six
+
+from pyarrow.compat import frombytes, tobytes
+from pyarrow.includes.common cimport *
+from pyarrow.includes.libarrow cimport *
+from pyarrow.includes.libarrow_s3 cimport *
+from pyarrow._fs cimport FileSystem
+from pyarrow.lib cimport check_status
+
+
+cdef class S3FileSystem(FileSystem):
+
+    cdef:
+        CS3FileSystem* s3fs
+
+    def __init__(self, str access_key, str secret_key, str region='us-east-1',
```

Comment:
I'd rather expose the S3Options struct in Python. Also, you should rebase as it has changed on the C++ side.

Reply:
Yep, I'm done with that locally.